apple/internal/docc.bzl

```diff
@@ -0,0 +1,244 @@
+)
+
+def _docc_archive_impl(ctx):
+    """Builds a .doccarchive for the given module.
+    """
+
+    default_code_listing_language = ctx.attr.default_code_listing_language
+    diagnostic_level = ctx.attr.diagnostic_level
+    enable_inherited_docs = ctx.attr.enable_inherited_docs
+    execution_requirements = {}
+    fallback_bundle_identifier = ctx.attr.fallback_bundle_identifier
+    fallback_bundle_version = ctx.attr.fallback_bundle_version
+    fallback_display_name = ctx.attr.fallback_display_name
+    kinds = ctx.attr.kinds
+    platform = ctx.fragments.apple.single_arch_platform
+    transform_for_static_hosting = ctx.attr.transform_for_static_hosting
+    xcode_config = ctx.attr._xcode_config[apple_common.XcodeVersionConfig]
+    dep = ctx.attr.dep[0]  # this isn't actually a list target but transition makes it one.
+    symbol_graphs_info = None
+    docc_bundle_info = None
+    docc_build_inputs = []
+
+    if DocCSymbolGraphsInfo in dep:
+        symbol_graphs_info = dep[DocCSymbolGraphsInfo]
+    if DocCBundleInfo in dep:
+        docc_bundle_info = dep[DocCBundleInfo]
+
+    if not symbol_graphs_info and not docc_bundle_info:
+        fail("At least one of DocCSymbolGraphsInfo or DocCBundleInfo must be provided for target %s" % ctx.attr.name)
+
+    if ctx.attr.name.endswith(".doccarchive"):
+        doccarchive_dir = ctx.actions.declare_directory(ctx.attr.name)
+    else:
+        doccarchive_dir = ctx.actions.declare_directory("%s.doccarchive" % ctx.attr.name)
+
+    # Command and required arguments
+    arguments = ctx.actions.args()
+    arguments.add("docc")
+    arguments.add("convert")
+    arguments.add("--index")
+    arguments.add("--fallback-display-name", fallback_display_name)
+    arguments.add("--fallback-bundle-identifier", fallback_bundle_identifier)
+    arguments.add("--fallback-bundle-version", fallback_bundle_version)
+    arguments.add("--output-dir", doccarchive_dir.path)
+
+    # Optional agruments
+    if default_code_listing_language:
+        arguments.add("--default-code-listing-language", default_code_listing_language)
+    if diagnostic_level:
+        arguments.add("--diagnostic-level", diagnostic_level)
+    if enable_inherited_docs:
+        arguments.add("--enable-inherited-docs")
+    if kinds:
+        arguments.add_all("--kind", kinds)
+    if transform_for_static_hosting:
+        arguments.add("--transform-for-static-hosting")
+
+    # Add symbol graphs
+    if symbol_graphs_info:
+        arguments.add_all("--additional-symbol-graph-dir", symbol_graphs_info.symbol_graphs, expand_directories = False)
+        docc_build_inputs.extend(symbol_graphs_info.symbol_graphs)
+
+    # The .docc bundle (if provided, only one is allowed)
+    if docc_bundle_info:
+        arguments.add(docc_bundle_info.bundle)
+
+        # TODO: no-sandbox seems to be required when running docc convert with a .docc bundle provided
+        # in the sandbox the tool is unable to open the .docc bundle.
```

Comment:
I assume this is likely expected (we do the same with some other Xcode tools IIRC), but do you remember what exactly was the error in this particular situation?

Reply:
I dont have the exact error but essentially the `.docc` bundle was missing from the sandbox so it would fail along the lines of "bundle not found"